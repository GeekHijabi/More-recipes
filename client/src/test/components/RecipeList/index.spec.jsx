import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedRecipes, { Recipes }
  from '../../../js/components/RecipeList';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


const props = {
  apiGetRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  recipes: [],
  search: [],
  pageCount: 1,
  isLoadingRecipe: true,
  searchValue: '',
  searchItem: jest.fn(() => Promise.resolve({ ok: true }))
};


describe('Recipes snapshot', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Recipes {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

it('should recieve next props', () => {
  const spy = sinon.spy(Recipes.prototype, 'componentWillReceiveProps');
  const nextProps = {
    apiGetRecipe: jest.fn(() => Promise.resolve({ ok: true })),
    recipes: [],
    search: [],
    pageCount: 1,
    isLoadingRecipe: true,
    searchValue: '',
    searchItem: jest.fn(() => Promise.resolve({ ok: true }))
  };

  const wrapper = shallow(<Recipes {...nextProps} />);
  wrapper.setProps({ prop: 12 });
  expect(spy.calledOnce).toEqual(true);
});

describe('onPageChange', () => {
  it('should change the page', () => {
    const current = {
      selected: 2
    };
    const wrapper = shallow(<Recipes {...props} />);
    wrapper.instance().onPageChange(current);
    expect(wrapper.instance().props.apiGetRecipe).toBeCalled();
  });
});

describe('Connected Recipes component', () => {
  it('component successfully rendered', () => {
    const store = mockStore({
      recipe: {
        recipes: [],
        search: {},
        pageCount: 0,
        isLoadingRecipe: true,
        searchValue: ''
      }
    });
    const wrapper = shallow(<ConnectedRecipes store={store} />);
    expect(wrapper.length).toBe(1);
  });
});
