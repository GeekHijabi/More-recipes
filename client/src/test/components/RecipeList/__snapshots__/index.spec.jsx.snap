// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes snapshot should render correctly 1`] = `
<div>
  <withRouter(Connect(RecipeHeader)) />
  <div
    style={
      Object {
        "margin": "0 auto",
        "minHeight": "795px",
        "width": "85%",
      }
    }
  >
    <div
      className="col-xs-12 col-sm-12 col-md-12 col-lg-12 headings"
    >
      <h4
        className="mb-4"
      >
        <span>
          Recipes
        </span>
      </h4>
    </div>
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <onlyUpdateForKeys(Loader)
          color="#B0C038"
          loading={true}
          size={60}
        />
        <div
          className="col-xs-12 col-sm-12 col-md-12 col-lg-12"
        >
          <span
            className="styleText"
          >
            No recipes yet, Come back soon for amazing recipes!
          </span>
        </div>
      </div>
    </div>
    <Connect(PopularRecipe) />
  </div>
  <Footer />
</div>
`;
